packages/foam-vscode/src/features/hover-provider.ts

```diff
@@ -108,8 +108,19 @@ export class HoverProvider implements vscode.HoverProvider {
         : this.workspace.get(targetUri).title;
     }
 
+    // If placeholder, offer to create a new note from template (compared to default link provider - not from template)
+    const command = OPEN_COMMAND.asURI(targetUri, true);
```

Comment:
Have you considered using the `CREATE_NOTE` command here? It would be "more in line" with the purpose of the action (somewhat related: I do think the `OPEN_COMMAND` does a bit too much in its current implementation, that was a design flaw back when I wrote it)

Reply:
Yeah, I think that makes more sense (ie. that's the point of the link here - to create a new note).

The create note command would just have to be modified so I can call it here in such a way that it asks the user for a template when run. I guess the command would either need an extra option to specify when it should ask the user for a template, or to also accept a function as the template argument, which would be called to get a template. Does that make sense / do you prefer either approach?

Reply:
It didn't work for me to pass a callback into `CREATE_NOTE_COMMAND.asURI` - it doesn't survive the `JSON.stringify`. So I guess an extra option is the best/only way?

Reply:
We should look into the `asUri` function, there must be an issue somewhere there. I can take a look later tomorrow or the day after, but if you see the problem beforehand go ahead and fix it